python/cudf/cudf/core/column/string.py

```diff
@@ -4637,9 +4640,30 @@ def _get_cols_list(others):
         If others is a list-like object (in our case lists & tuples)
         just another Series/Index, great go ahead with concatenation.
         """
-        cols_list = [column.as_column(frame, dtype="str") for frame in others]
+
+        cols_list = []
+
+        for frame in others:
+            if isinstance(frame, cudf.Index):
+                frame = frame.to_series()
```

Comment:
This is actually not needed, because this issue was fixed: https://github.com/pandas-dev/pandas/issues/33425
But after some digging the inverse operation still fails in pandas:
```python
sr = pd.Series(['a','b','c','d','e'])
others = pd.Index(['a','b','c','d','e'])
print(sr.str.cat(others))
print(others.str.cat(sr))

0    aa
1    bb
2    cc
3    dd
4    ee
dtype: object
Index([nan, nan, nan, nan, nan], dtype='object')
```

Previous code was trying to match the inverse behaviour. Once this [issue](https://github.com/pandas-dev/pandas/issues/35556) is fixed we can uncomment the xfail tests.